### PR TITLE
vsphere support

### LINKED ISSUE #277: Support VMware vSphere
The machine-controller https://github.com/kubermatic/machine-controller/tree/master/pkg/cloudprovider/provider/vsphere already support vSphere. We could add support here as well

### PR DESCRIPTION
**Pull request by kron4eg**
Fixes #277

```release-note
vSphere support
```

### CODE REVIEW

**Comment by kron4eg** [comment]
My local vars was:

```
cluster_name = "artiom42"
datastore_name = "exsi-nas"
network_name = "VM Network"
template_name = "machine-controller-e2e-ubuntu"
```

**Comment by kdomanski** [comment]
You cannot really do OS customization through vsphere API for the templates we use. We don't even do that in machine-controller, we use metadata or an ISO depending on the OS.

After removing the `customize { ... }` section it is working fine for me. @kron4eg 

See also:
 - https://github.com/terraform-providers/terraform-provider-vsphere/issues/388#issuecomment-427357809
 - https://github.com/vmware/open-vm-tools/issues/299

**Comment by kron4eg** [comment]
OH... That'd explain it. I'll try another time, thanks!

File: docs/quickstart-vsphere.md [hunk 1]
```diff
@@ -0,0 +1,294 @@
...
+to destroy the cluster along with the infrastructure.
+
+As a result, you'll get Kubernetes 1.14.1 High-Available (HA) clusters with
+three control plane nodes and two worker nodes.
+
+### Prerequisites
+
+To follow this quick start, you'll need:
+
+* `kubeone` v0.7.0 or newer installed, which can be done by following the
```
Comment (xmudrii) [inline docs/quickstart-vsphere.md]: Actually, you need `master` or `v0.8.0`, because `v0.7.0` doesn't support vSphere.

Comment (kron4eg) [reply docs/quickstart-vsphere.md]: 👍 done

**Comment by xmudrii** [comment]
/lgtm
/approve

:tada: :100:

**Comment by toschneck** [comment]
Woohooo great job guys! @thetechnick is now ready for your setup 🥰

**Pull request merged**